lib/datasource.js

```diff
@@ -632,6 +632,7 @@ DataSource.prototype.setupDataAccess = function(modelClass, settings) {
       if (idProp.generated && forceId !== false) {
         forceId = true;
       }
+      settings.forceId = forceId;
```

Comment:
I believe the lines 632-635 can be removed now, because `forceId` is already calculated in `lib/model-builder`. Unless I am missing something?